Core.lua

```diff
@@ -589,9 +589,16 @@ do
 	-- @name IsRowVisible
 	-- @description Checks if a row is currently being shown
 	-- @usage st:IsRowVisible(realrow)
-	-- @thanks sapu94
 	local function IsRowVisible(self, realrow)
-		return (realrow > self.offset and realrow <= (self.displayRows + self.offset))
+		-- Scan through all on-screen rows, checking their real data row numbers for a match.
+		local firstVisibleIdx = 1 + (self.offset or 0);
+		local lastVisibleIdx = (firstVisibleIdx + self.displayRows) - 1;
+		for i=firstVisibleIdx,lastVisibleIdx do
+			if self.filtered[i] == realrow and realrow ~= nil then
```

Comment:
This line, explained: (I mentioned this explanation at wowace where we began our discussions)

The scrollingtable library has two arrays. `.data` which is the actual data, and `.filtered` which is a sorted array of what `.data` table index to show on each visual row, such as `{1, 5, 4, 6, 2...}`.

The user wants to call `IsRowVisible()` to know if a given `.data` row is visible on screen.

So we're checking if `self.filtered[i] == realrow` to see if that on-screen row points at the given data row they're checking for...

And then, if we found a match, we do a final, quick `and realrow ~= nil` to ensure they didn't provide our function with a nil `realrow` value. This small failsafe is necessary to weed out the scenario where the user has failed to provide any `realrow` value, and where some of the on-screen table rows don't contain any data (such as if there are 12 rows on screen, but only 6 entries in `.filtered`, in which case the final entries (7-12) being checked will all contain `nil`).

This failsafe could also have been written as `if (realrow == nil) then error("Must provide non-nil realrow value."); end` or something like that at the top of the function, but I decided to do it this way instead just to avoid throwing errors, and to avoid putting an extra line of code at the top of the function. It's just a stylistic choice to do the check here instead of at the top... and it has some benefits (shorter code, less function return-paths...).